### PR TITLE
Update build-common.groovy: add rust code cache for tiflash

### PR DESCRIPTION
**Pull request by solotzg**

### CODE REVIEW

File: atom-jobs/build-common.groovy [hunk 1]
```diff
@@ -11,7 +11,7 @@
 * @FORCE_REBUILD(bool:if force rebuild binary,default true,Optional)
 * @FAILPOINT(bool:build failpoint binary or not,only for tidb,tikv,pd now ,default false,Optional)
 * @EDITION(enumerate:,community,enterprise,Required)
-* @UPDATE_TIFLASH_CACHE(bool: update ci build cache, for tiflash only, default false, Optional)
```
Comment (solotzg) [inline atom-jobs/build-common.groovy:14]: `UPDATE_TIFLASH_CACHE` is no longer used.

File: atom-jobs/build-common.groovy [hunk 2]
```diff
@@ -523,6 +519,20 @@ if [ ${OS} == 'darwin' ]; then
     ls -l ./release-darwin/tiflash/
     mv release-darwin ${TARGET}
 else
+    if [[ "${USE_TIFLASH_RUST_CACHE}" == "true" ]]; then
```
Comment (purelind) [inline atom-jobs/build-common.groovy:522]: USE_TIFLASH_RUST_CACHE us a booleanParam,  properties need set this param

Comment (solotzg) [reply atom-jobs/build-common.groovy:527]: fixed

**Review by SchrodingerZhu — APPROVED** [review]

**Comment by solotzg** [comment]
After tests:
- set `USE_TIFLASH_RUST_CACHE` to `true`: https://cd.pingcap.net/blue/organizations/jenkins/build-common/detail/build-common/87733/pipeline.
- unset `USE_TIFLASH_RUST_CACHE`: https://cd.pingcap.net/blue/organizations/jenkins/build-common/detail/build-common/87734/pipeline/

Both work as expected.

**Comment by solotzg** [comment]
Also need to modify: https://github.com/PingCAP-QE/ci/pull/840

**Review by purelind — APPROVED** [review]

**Pull request merged**